### PR TITLE
[IR] Refactor scalar attribute

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

New features
### PR changes

Others
### Description

This PR actually deletes `paddle::diaclect::ScalarAttribute` and make it unusable in most cases. But we allow several basic builtin attributes to be identified as a `paddle::diaclect::ScalarAttribute`.

#### Others
Pcard-67164

### CODE REVIEW

**Review by winter-wang — DISMISSED** [review]
LGTM

**Review by zhangbo9674 — DISMISSED** [review]
LGTM

File: paddle/fluid/ir/dialect/utils.h [hunk 1]
```diff
@@ -70,6 +72,30 @@ inline ir::Type TransToIrDataType(phi::DataType dtype,
   }
 }
 
+static inline ir::Attribute TransToIrAttribute(phi::Scalar scalar,
```
Comment (Shixiaowei02) [inline paddle/fluid/ir/dialect/utils.h:75]: 取消内联，实现放入 `.cc` 可能好一些

**Review by Shixiaowei02 — DISMISSED** [review]

File: paddle/fluid/ir/dialect/utils.h (hunk 1 above)
Comment (kangguangli) [reply paddle/fluid/ir/dialect/utils.h:75]: 有道理，后面我单独提个PR重构下。

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Pull request merged**